### PR TITLE
feat/QB-1228 parallel download for RPC

### PR DESCRIPTION
**Pull request by WangLiStratos**
This PR is on top of PR #192 (feat/QB-1199 RPC download file). The first two commits are from #192. When #192 is merged, this branch will be rebased.

* ReqId of the file is used for identifying the download sessions
* LOCAL_REQID is used for local download, and a generated uuid is used for remote download
* keep a map to match slice ReqId to a file
* different way to handle the end of download - no local tem file nor csv file for remote download
* ReqId included in download related message

### CODE REVIEW

File: pp/api/rpc/rpc_api.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	FileSize      uint64   `json:"filesize"`
+	ReqId         string   `json:"reqid"`
+}
+
+// result for all upload and download messages
+type Result struct {
+	Return        string     `json:"return"`
+	ReqId         string     `json:"reqid,omitempty"`
+	OffsetStart   *uint64    `json:"offsetstart,omitempty"`
+	OffsetEnd     *uint64    `json:"offsetend,omitempty"`
```
Comment (alexstratos) [inline pp/api/rpc/rpc_api.go:61]: better not use pointer

File: pp/serv/rpc.go [hunk 2]
```diff
@@ -25,10 +25,9 @@ const (
 	FILE_DATA_SAFE_SIZE = 3500000
 
 	// timeout for waiting result from external source, in seconds
-	WAIT_TIMEOUT time.Duration = 3
+	WAIT_TIMEOUT time.Duration = 5
```
Comment (alexstratos) [inline pp/serv/rpc.go:28]: why not directly use time.Second or time.MilliSecond

File: pp/api/rpc/rpc_api.go (hunk 1 above)
Comment (WangLiStratos) [reply pp/api/rpc/rpc_api.go:61]: The simple uint64 type don't have an "empty" value to be omitted when marshal this struct. Pointers here should be safe I think.

File: pp/serv/rpc.go (hunk 2 above)
Comment (WangLiStratos) [reply pp/serv/rpc.go:28]: Good point. Changing it.

**Review by alexstratos — APPROVED** [review]

**Pull request merged**